src/views/Home/components/CLWCreditsDaysLeft/index.js

```diff
@@ -11,194 +11,192 @@ import './CLWChart.css';
 
 import { Card, CardHeader, CardContent, Typography, Grid, Button } from '@material-ui/core';
 
-export default class CLWCreditsDaysLeft extends Component {
-  constructor(props) {
-    super(props);
-
-    this.loadData = this.loadData.bind(this);
-
-    this.state = {
-      daysLeft: [],
-      chapelCredits: {},
-      error: null,
-      loading: true,
-      currSessionDescription: '',
+const CLWCreditsDaysLeft = () => {
+  const [state, setState] = useState({
+    daysLeft: [],
+    chapelCredits: {},
+    error: null,
+    loading: true,
+    currSessionDescription: '',
+  });
```

Comment:
This doesn't work because of one of the key differences between `React.Component.setState` and `useState`. 

The former did shallow object merging, so if `state` was an object with properties 
```
{
    daysLeft: [],
    chapelCredits: {},
    error: null,
    loading: true,
    currSessionDescription: '',
}
```
and you called `setState({ loading: false })`, then it would replace `state.loading` with the new value `true`, leaving `state` as 
```
{
    daysLeft: [],
    chapelCredits: {},
    error: null,
    loading: false,
    currSessionDescription: '',
}
```


The new hook `useState` doesn't do any kind of merging - it will replace the previous state with the new state. What that means is that if you had the same starting `state` of 
```
{
    daysLeft: [],
    chapelCredits: {},
    error: null,
    loading: true,
    currSessionDescription: '',
}
``` 
and then called the `setState` from the new hook as `setState({ loading: false })`, then your new state will simply be
```
{
    loading: false
}
```

The fix for this is to separate the different parts of the component's state into separate variables:
```
const [daysLeft, setDaysLeft] = useState([])
const [chapelCredits, setChapelCredits] = useState({})
...
```